### DISCUSSION TITLE
Deploying to Azure with an Extension/Custom Endpoint

### DISCUSSION THREAD

**Discussion by chantlong** [directus/directus#14617] [Archive]
I followed the following tutorial to deploy Directus to Azure.
https://devblogs.microsoft.com/azure-sql/automatic-graphql-and-rest-endpoint-for-azure-sql-with-directus/

**But I want to do deploy a custom extension. How would you do that?**

I'm guessing follow the Manual Installation...
https://docs.directus.io/self-hosted/installation/manual/

and then like [GCP](https://docs.directus.io/self-hosted/installation/gcp/) create a Dockerfile with

```
FROM node:16-alpine
WORKDIR /src
ADD . /src
RUN npm install --production
CMD ["npm", "run", "start"]
```

Then deploy to Azure's Container Registry and have App Service reference that container instead?

Thanks

**Comment by azrikahar** [comment]
That does sound like a plausible approach 👍

Depending on the preferred workflow/requirement, an alternative may be to use [App Service's `Deployment Center`](https://docs.microsoft.com/en-us/azure/app-service/deploy-continuous-deploymen) and use your repo as the sourc.
